include/vcpu/arch/intel_x64/vcpu.h

```diff
@@ -393,18 +418,33 @@ class vcpu : public bfvmm::intel_x64::vcpu
 
     void check_wrcr8()
     {
-        check_crall();
+        init_control_register();
 
         if (!m_is_wrcr8_enabled) {
             m_is_wrcr8_enabled = true;
             m_control_register->enable_wrcr8_exiting();
         }
     }
 
-    void check_monitor_trap()
+    void init_cpuid()
+    {
+        if (!m_cpuid) {
+            m_cpuid = std::make_unique<eapis::intel_x64::cpuid>(this->exit_handler());
+            m_handlers[vmcs_n::exit_reason::basic_exit_reason::cpuid] = m_cpuid.get();
```

Comment:
I'd rather this be a list, and not a hash. Thoughts? The list will be cleaner on memory, but the hash is cleaner to program, and faster. Just depends on how much memory the hash consumes for this.